mrmustard/physics/ansatze.py

```diff
@@ -380,3 +380,114 @@ def __and__(self, other: PolyExpAnsatz) -> PolyExpAnsatz:
         bs = [math.concat([b1, b2], axis=-1) for b1 in self.b for b2 in other.b]
         cs = [math.outer(c1, c2) for c1 in self.c for c2 in other.c]
         return self.__class__(As, bs, cs)
+
+class ArrayAnsatz(Ansatz):
+    r"""Represents the ansatz as a multidimensional array.
+    """
+    def __init__(self, array: Batch[Tensor]):
+        r"""Note that the array is batched."""
+        self.array = array
+    
+    @property
+    def array(self) -> Batch[Tensor]:
+        r"""
+        The array.
+        """
+        return self.array
+
+    def __neg__(self) -> ArrayAnsatz:
+        r"""
+        Negates the values in the array.
+        """
+        return self.__class__(array=-self.array)
+
+    def __eq__(self, other: Ansatz) -> bool:
+        r"""
+        Whether this ansatz's array is equal to another ansatz's array.
+
+        Note that the comparaison is done by numpy allclose with numpy's default rtol and atol.
+
+        #TODO: make sure this is the correct type of the wrong shape
+        """
+        try:
+            return np.allclose(self.array, other.array)
+        except Exception as e:
+            raise TypeError(f"Cannot add {self.__class__} and {other.__class__}.") from e
+
+    def __add__(self, other: ArrayAnsatz) -> ArrayAnsatz:
+        r"""
+        Adds the array of this ansatz and the array of another ansatz.
+
+        Args:
+            other: Another ansatz.
+
+        Raises:
+        #TODO: make sure this is the correct type of the wrong shape
+            TypeError: If both of array don't have the same shape.
+
+        Returns:
+            ArrayAnsatz: The addition of this ansatz and other.
+        """
+        try:
+            return self.__class__(array = self.array + other.array)
+        except Exception as e:
+            raise TypeError(f"Cannot add {self.__class__} and {other.__class__}.") from e
+
+    def __call__(self, point: Any) -> Scalar:
+        r"""
+        Evaluates this ansatz at a given point in the domain.
+        """
+        raise AttributeError("Cannot plot ArrayAnsatz.")
```

Comment:
Cannot **plot**?

Reply:
Don't know how to plot a Fock representation, but we can do it in Bargmann. Then it is an abstract method I have to implement. So I raise this Error here...

Reply:
Maybe we could plot it as a histogram, where you have `n` on the x axis and `abs(array[n])` on the y axis?
Not saying we do it in this PR, but conceptually visualizing arrays should be easier than visualizing A, B, C triples, so I think we can find a way :)

Reply:
Call was to get the value of the ansatz f(x), which made sense for Bargmann, because it's continuous. Here f(x) is like self.array[x] so x are the lattice points, but I don't know if it's just better to not implement call()